src/views/primary-auth/PrimaryAuthForm.js

```diff
@@ -119,7 +120,11 @@ define([
         input: TextBox,
         inputId: 'okta-signin-username',
         type: 'text',
-        disabled: this.options.appState.get('disableUsername')
+        disabled: this.options.appState.get('disableUsername'),
+        // TODO: support a11y attrs in Courage - OKTA-279025
+        render: function () {
+          this.$(`input[name=${this.options.name}]`).prop('required', true);
```

Comment:
Do we know if the `required` attribute is supported by all browsers ? SIW gets initialized inside embedded browsers in some cases.

Reply:
Some comments:
- isn't the name always `username`? As in, we probably don't need to get it from `this.options.name`... or even `this.$el.prop('required', true);` could work
- should this be done in `postRender`? 
- Can we wait until OKTA-279025 is done? Instead of doing a one-off here, we could wait for courage to support this... it might take more time, so I'm fine if we don't want to wait, but wanted to ask just in case

Reply:
> Do we know if the `required` attribute is supported by all browsers ? SIW gets initialized inside embedded browsers in some cases.

@nikhilvenkatraman-okta  As `novalidate` attribute is added to the form, validations are still depend on our own logic. If `required` is not supported, it will just be ignored by browser silently.

Reply:
> * isn't the name always `username`? As in, we probably don't need to get it from `this.options.name`... or even `this.$el.prop('required', true);` could work
Looks like the input actually is a `div` container wraps `input`, so `this.$el.prop('required', true);` will only apply `required` to the container `div`. 
Per use the ref from `this.options.name`, it can free us from potential attr change, even by myself I don't really think it may happen :)

> * should this be done in `postRender`?
As I mentioned in the question above, the model refer to the input container instead of the input element, I still think a post render is needed here.

> * Can we wait until OKTA-279025 is done? Instead of doing a one-off here, we could wait for courage to support this... it might take more time, so I'm fine if we don't want to wait, but wanted to ask just in case
Good question, probably my project manager @chrismorris-okta can help answer this question better. IMO, as the changes are already ready, the engineer effort of waiting for `courage` support would not differ too much. But roll it out can make devs access to this a11y feature earlier.

Reply:
sounds good about 1 and 3
about the `postRender`, I meant if this should be done in the `postRender` method instead of overriding `render`

Reply:
> sounds good about 1 and 3
> about the `postRender`, I meant if this should be done in the `postRender` method instead of overriding `render`

Good catch! Didn't realize `postRender` hook been provided.

Reply:
@mauriciocastillosilva-okta After debugging `postRender` is not available for inputs in `PrimaryAuthForm`. Let's keep using `render` here.

Reply:
it should be supported. The input we are using is a `Textbox` which we already override in [views/shared/Textbox](https://github.com/okta/okta-signin-widget/blob/master/src/views/shared/TextBox.js) and we are using a `postRender` function there... I can take a look later today see if I can figure this out.

Reply:
talked offline. This works fine: `render` here is called inside the `postRender` of the input, which is confusing.
We will create a follow up jira to change this later, either when OKTA-279025 is done (courage support for `required`) or when we more inputs to support `required`.

Reply:
Ticket to handle render method: https://oktainc.atlassian.net/browse/OKTA-280596